verification/cpp/suite/test_unionant.cpp

```diff
@@ -66,22 +66,69 @@ TEST(UnionantTests, get_set_lvalue)
     }
 }
 
+/**
+ * Verify that the variant value can be fetched only as the type being held
+ */
 TEST(UnionantTests, get_if_const_variant)
 {
     using ValueType = uavcan::_register::Value_1_0;
-    const uavcan::_register::Value_1_0             a{};
-    const uavcan::primitive::array::Integer32_1_0* p =
+    const ValueType a{};
+
+    const uavcan::primitive::Empty_1_0* p_empty =
+        uavcan::_register::Value_1_0::VariantType::get_if<ValueType::VariantType::IndexOf::empty>(&a.union_value);
+    const uavcan::primitive::array::Integer32_1_0* p_int32 =
         uavcan::_register::Value_1_0::VariantType::get_if<ValueType::VariantType::IndexOf::integer32>(&a.union_value);
-    ASSERT_EQ(nullptr, p);
+
+    ASSERT_NE(nullptr, p_empty);
+    ASSERT_EQ(nullptr, p_int32);
 }
 
+/**
+ *  Verify that the variant value can be fetched only at the alternative index for the type being held
```

Comment:
Missing a test that actually passes an allocator through to a variant type constructor.